### DISCUSSION TITLE
写真を撮った場所にピンを置きたい

### DISCUSSION THREAD

**Discussion by Keita0920** [lit-iphone-course/lit-iphone-course#39] [Question-original] [answered]
(1) 実現したいこと
カメラで写真を撮ると撮った場所の位置情報(緯度経度)をRealmに保存し、写真をこのアプリ内のドキュメントの中に保存、その写真を読み込むためのファイルURLをRealmに保存。最終的にマップに写真がたくさん表示されるようにしたい。

(2) 実現するために自分が試した内容とその結果
1.緯度と経度を取得し代入しようとするとエラー文→No exact matches in call to initializer
2.写真を撮るときにその場所にピンをさして、そのピンの画像を撮った写真をリサイズしたものに差し替えたいけれども画像が挿入されない(多分ピン自体が立っていないのだけれどもその原因がわからない)

(3) 調べた内容や資料
1.https://qiita.com/jpmos7/items/e32570c8acb0282b6c23
2.https://faboplatform.github.io/SwiftDocs/3.mapkit/008_image/

(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
        updateCurrentPos((locations.last?.coordinate)!)
        guard let loc = locations.last else { return }
        self.loc=loc
        self.latitude=Double(String(locations.first?.coordinate.latitude))!
        self.longitude=Double(String(locations.first?.coordinate.longitude))!
        CLGeocoder().reverseGeocodeLocation(loc, completionHandler: {(placemarks, error) in
            if let error = error {
                print("reverseGeocodeLocation Failed: \(error.localizedDescription)")
                return
            }
        })
    }

(5) GitHubのリンク
https://github.com/Keita0920/LA_RePlace

**Comment by txxxxc** [comment, marked as answer]
`No exact matches in call to initializer`が出る原因は多分末尾の`!`が原因だと思います！
あと`locations`には最新の位置情報が配列の末尾に格納されていくはず(検証してないので間違ってたらごめんなさい)なので、`locations.first`ではなく`locations.last`を使ってあげましょう！！

具体的にはこんな感じで書くとうまくいくと思います！
```
        guard let loc = locations.last else { return }
        self.loc=loc
        self.latitude=Double(loc.coordinate.latitude)
        self.longitude=Double(loc.coordinate.longitude)
```
あと、ピンがうまく刺さらないエラーに関しては別途質問をと建ててもらえるとありがたいです！そのほうが後で見返しやすいので！！
お手数ですがご協力お願いします！！

**Reply by txxxxc** [reply]
お疲れ様です！こちらの問題解決出来たでしょうか..?
ヤバそうだったら教えてもらえると嬉しいですー！
